src/main/java/graphql/schema/idl/SchemaGenerator.java

```diff
@@ -257,13 +259,66 @@ private GraphQLObjectType buildObjectType(BuildContext buildCtx, ObjectTypeDefin
         builder.name(typeDefinition.getName());
         builder.description(buildDescription(typeDefinition));
 
-        typeDefinition.getFieldDefinitions().forEach(fieldDef ->
-                builder.field(buildField(buildCtx, typeDefinition, fieldDef)));
+        //
+        // fields
+        Map<String, GraphQLFieldDefinition> fieldDefinitions = new LinkedHashMap<>();
+
+        typeDefinition.getFieldDefinitions().forEach(fieldDef -> {
+            GraphQLFieldDefinition newFieldDefinition = buildField(buildCtx, typeDefinition, fieldDef);
+            fieldDefinitions.put(newFieldDefinition.getName(), newFieldDefinition);
+        });
+
+        //
+        // type extensions feed into the fields of an object type
+        List<TypeExtensionDefinition> typeExtensions = getTypeExtensionsOf(typeDefinition, buildCtx);
+
+
+        typeExtensions.forEach(typeExt -> typeExt.getFieldDefinitions().forEach(fieldDef -> {
+            GraphQLFieldDefinition newFieldDefinition = buildField(buildCtx, typeDefinition, fieldDef);
+            //
+            // de-dupe here - pre-flight checks ensure all dupes are of the same type
+            if (!fieldDefinitions.containsKey(newFieldDefinition.getName())) {
+                fieldDefinitions.put(newFieldDefinition.getName(), newFieldDefinition);
+            }
+        }));
+
+        // combine all the fields together
+        for (GraphQLFieldDefinition fieldDefinition : fieldDefinitions.values()) {
+            builder.field(fieldDefinition);
+        }
+
+        //
+        // interfaces
```

Comment:
I don't think such comments are very helpful: If you wanna document the following block I suggest extracting a method and name the method appropriately.